lib/install/validate/validator.go

```diff
@@ -559,107 +558,108 @@ func (v *Validator) registries(ctx context.Context, input *data.Data, conf *conf
 	}
 }
 
-func (v *Validator) friendlyRegistryList(registryType string, registryList []url.URL) string {
-	var msg string
-
-	if len(registryList) > 0 {
-		var host string
-		msg = registryType + " = "
-		for i, r := range registryList {
-			host = r.Host
-			if strings.Contains(r.Path, "/") {
-				host = host + r.Path
-			}
-
-			if i == 0 {
-				msg = msg + host
-			} else {
-				msg = msg + ", " + host
-			}
-		}
+func (v *Validator) friendlyRegistryList(registryType string, registryList []string) string {
+	if len(registryList) == 0 {
+		return ""
 	}
 
-	return msg
+	return registryType + " = " + strings.Join(registryList, ", ")
 }
 
 // Validate registries are reachable.  Secure registries that are not specified as insecure are validated with the
 // CA certs passed into vic-machine.
-func (v *Validator) reachableRegistries(ctx context.Context, input *data.Data, pool *x509.CertPool) (insecureRegistries []url.URL, whitelistRegistries []url.URL, err error) {
-	secureRegistries := input.WhitelistRegistries
-
-	// Remove intersection between insecure registries and whitelist registries from whitelist set so
-	// we can ensure we test the exclusion set with certs
-	var exist bool
-	var idx int
-	var s url.URL
-	for _, r := range input.InsecureRegistries {
-		exist = false
-		for idx, s = range secureRegistries {
-			if r.Host == s.Host {
-				exist = true
-				break
-			}
-		}
+func (v *Validator) reachableRegistries(ctx context.Context, input *data.Data, pool *x509.CertPool) (insecureRegistries []string, whitelistRegistries []string, err error) {
+	secureRegistriesSet, err := dynamic.ParseRegistries(input.WhitelistRegistries)
+	if err != nil {
+		return nil, nil, err
+	}
 
-		// remove the insecure registry from list of registries to get validated against certs
-		if exist {
-			secureRegistries = append(secureRegistries[:idx], secureRegistries[idx+1:]...)
-		}
+	insecureRegistriesSet, err := dynamic.ParseRegistries(input.InsecureRegistries)
+	if err != nil {
+		return nil, nil, err
 	}
 
 	// Test insecure registries' reachability
-	for _, r := range input.InsecureRegistries {
+	for _, r := range insecureRegistriesSet {
+		r, ok := r.(registry.URLEntry)
+		if !ok {
+			err = fmt.Errorf("invalid insecure registry entry: %s", r)
```

Comment:
The only case we can get here is if there is a CIDREntry. That is not practically possible since we catch those entries in vic-machine. If there is an invalid URL, `dynamic.ParseRegistries` will return an error.